src/ray/core_worker/core_worker.cc

```diff
@@ -557,6 +557,18 @@ CoreWorker::CoreWorker(const CoreWorkerOptions &options, const WorkerID &worker_
       new CoreWorkerDirectActorTaskSubmitter(core_worker_client_pool_, memory_store_,
                                              task_manager_));
 
+  pubsub_coordinator_ = std::shared_ptr<PubsubCoordinator>(new PubsubCoordinator(
```

Comment:
`make_shared<Foo>()` performs one heap allocation while `std::shared_ptr<Foo>(new Foo())` performs two heap allocations.

Reply:
`make_shared` makes the counter and object live in the same block. As mentioned, above it avoids one extra heap application, besides, since they are adjacent, it's also cache friendly.